tools/cli/commands/clean.js

```diff
@@ -0,0 +1,533 @@
+
+/**
+ * Command definition for the build subcommand.
+ *
+ * @param {object} yargs - The Yargs dependency.
+ *
+ * @returns {object} Yargs with the build commands defined.
+ */
+export function cleanDefine( yargs ) {
+	yargs.command(
+		'clean [project] [include]',
+		'Removes unversioned files and folder from a specific project.',
+		yarg => {
+			yarg
+				.positional( 'project', {
+					describe: 'Project in the form of type/name, e.g. plugins/jetpack, or type, e.g. plugins, or "all"',
+					type: 'string',
+				} )
+				.positional( 'include', {
+					alias: 'i',
+					describe: 'Files/folders to include for deletion',
+					type: 'string',
+					choices: [ 'untracked', 'ignored', 'docker', 'node_modules', 'composer.lock', 'vendor' ],
+				} )
+				.option( 'all', {
+					alias: 'a',
+					type: 'boolean',
+					description: 'Clean all types of files everywhere in the monorepo',
+				} )
+				.option( 'dist', {
+					type: 'boolean',
+					description: 'Remove package manager directories (vendor, node_modules)',
+				} );
+		},
+		async argv => {
+			await cleanCli( argv );
+			if ( argv.v ) {
+				console.log( argv );
+			}
+		}
+	);
+
+	return yargs;
+}
+
+/**
+ * Handle args for clean command.
+ *
+ * @param {argv}  argv - the arguments passed.
+ */
+export async function cleanCli( argv ) {
+	if ( argv.all ) {
+		argv.project = '.';
+		argv.include = [ 'untracked', 'ignored', 'docker', 'node_modules', 'composer.lock', 'vendor' ];
+	}
+
+	if ( argv.dist ) {
+		argv.project = '.';
+		argv.include = [ 'node_modules', 'vendor' ];
+	}
+
+	if ( argv.project ) {
+		argv = await parseProj( argv );
+	} else {
+		argv = await promptProj( argv );
+	}
+
+	if ( argv.include ) {
+		await parseToClean( argv );
+	} else {
+		await promptForClean( argv );
+	}
+
+	// Collect the files we want to clean.
+	const allFiles = await collectAllFiles( argv.toClean, argv );
+	const toCleanFiles = await collectCleanFiles( allFiles, argv.toClean );
+
+	// Bail if there are no files to delete.
+	if ( ! toCleanFiles.length ) {
+		console.log( chalk.green( 'No files to delete!' ) );
+		return;
+	}
+
+	// Confirm the deletion.
+	const runConfirm = await confirmRemove( argv, toCleanFiles );
+	if ( ! runConfirm.confirm ) {
+		console.log( chalk.red( 'Cancelling jetpack clean.' ) );
+		return;
+	}
+
+	await cleanFiles( toCleanFiles, argv );
+	return;
+}
+
+/**
+ * Delete the files that we want.
+ *
+ * @param {Array} toCleanFiles - files that we want to clean.
+ * @param {object} argv - the arguments passed.
+ */
+async function cleanFiles( toCleanFiles, argv ) {
+	for ( const file of toCleanFiles ) {
+		console.log( `Cleaning ${ file }` );
+		try {
+			fs.rmSync( file, { recursive: true, force: true } );
+		} catch ( e ) {
+			// Although it looks like rmSync doesn't do any error reporting, it probably should.
+			console.error( chalk.red( e.message ) );
+			process.exit( 1 );
+		}
+	}
+
+	// Cleanup root and tools node_modules folder if that's what we're deleting.
```

Comment:
Personally I'd be a bit more explicit here. But feel free to ignore if you disagree.
```suggestion
	// The `esm` module has an 'exit' handler that will re-create some node_modules dirs to save its cached data.
	// Register our own 'exit' handler to re-delete them after esm re-creates them, when applicable.
```